### PR TITLE
#4364 Fix table reader - show better error message for bad selector

### LINKED ISSUE #4364: Table Reader Brick Errors with `e.rows is not iterable`
Context
---
- Tested with 1.7.9-beta.1
- Page:  https://mthek10.github.io/OldPortal/index.html

Discussion
---
- I think the error message might need to be changed to "Element is not a table" or something similar?

image

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Fixes #4364 

## Checklist

- [ ] Add tests
- [x] Designate a primary reviewer - @twschiller

### CODE REVIEW

File: src/blocks/transformers/component/TableReader.ts [hunk 1]
```diff
@@ -85,15 +86,21 @@ export class TableReader extends Transformer {
...
       return parseDefinitionList(table);
     }
 
-    return parseDomTable(table, { orientation: args.orientation });
+    if (table instanceof HTMLTableElement) {
+      return parseDomTable(table, { orientation: args.orientation });
+    }
+
+    throw new TypeError(
+      `Selector does not match a table or list element, found: <${lowerCase(
```
Comment (twschiller) [inline src/blocks/transformers/component/TableReader.ts:100]: It should read "definition list". A definition list is the `dl` tag which is distinct from ul/ol

**Review by twschiller — APPROVED** [review]
Conditionally improved. See message correction

**Review by twschiller — APPROVED** [review]
Thanks! 💯

File: src/blocks/transformers/component/TableReader.ts [hunk 2]
```diff
@@ -85,15 +86,21 @@ export class TableReader extends Transformer {
...
       return parseDefinitionList(table);
     }
 
-    return parseDomTable(table, { orientation: args.orientation });
+    if (table instanceof HTMLTableElement) {
+      return parseDomTable(table, { orientation: args.orientation });
+    }
+
+    throw new TypeError(
+      `Selector does not match a table or definition list (dl) element, found: <${lowerCase(
```
Comment (fregante) [inline src/blocks/transformers/component/TableReader.ts:100]: .toLowerCase() is a native JavaScript function, we don't need lodash here

**Pull request merged**